### PR TITLE
Remove hierarchical storage manager

### PR DESCRIPTION
**Pull request by eyusufatik**
Removes hierchical storage manager. so we get a cleaner stf blueprint usage. (and many more changes that make so much sense)

### CODE REVIEW

File: crates/sovereign-sdk/module-system/sov-modules-core/tests/working_set_tests.rs [hunk 1]
```diff
@@ -18,14 +12,14 @@ fn test_workingset_get() {
...
-    let mut working_set = WorkingSet::<DefaultContext>::new(storage.clone());
+    let mut working_set = WorkingSet::new(storage.clone());
     working_set.set(&storage_key, storage_value.clone());
 
     assert_eq!(Some(storage_value), working_set.get(&storage_key));
 }
 
 #[test]
-fn test_versioned_workingset_get() {
+fn test_workingset_get2() {
```
Comment (yaziciahmet) [inline crates/sovereign-sdk/module-system/sov-modules-core/tests/working_set_tests.rs]: exact same test with test_workingset_get()

File: crates/sovereign-sdk/module-system/sov-modules-api/src/lib.rs [hunk 2]
```diff
@@ -213,7 +212,7 @@ pub mod da {
 }
 
 pub mod storage {
```
Comment (yaziciahmet) [inline crates/sovereign-sdk/module-system/sov-modules-api/src/lib.rs]: Completely removable

File: crates/sovereign-sdk/full-node/sov-prover-storage-manager/src/lib.rs
Comment (yaziciahmet) [inline crates/sovereign-sdk/full-node/sov-prover-storage-manager/src/lib.rs:1]: Remove `HierarchicalStorageManager` from the doc of ProverStorageManager

File: crates/evm/src/query.rs [hunk 3]
```diff
@@ -1759,7 +1768,7 @@ fn update_estimated_gas_range(
 #[inline]
 fn set_state_to_end_of_evm_block<C: sov_modules_api::Context>(
```
Comment (yaziciahmet) [inline crates/evm/src/query.rs:1760]: generic can be storage instead of context here

File: crates/evm/src/lib.rs [hunk 4]
```diff
@@ -172,15 +172,19 @@ impl<C: sov_modules_api::Context> sov_modules_api::Module for Evm<C> {
 
     type Event = ();
 
-    fn genesis(&self, config: &Self::Config, working_set: &mut WorkingSet<C>) {
+    fn genesis(
+        &self,
+        config: &Self::Config,
+        working_set: &mut WorkingSet<<Self::Context as Spec>::Storage>,
```
Comment (yaziciahmet) [inline crates/evm/src/lib.rs]: just use C as Spec

File: crates/evm/src/lib.rs [hunk 5]
```diff
@@ -172,15 +172,19 @@ impl<C: sov_modules_api::Context> sov_modules_api::Module for Evm<C> {
...
+    ) {
         self.init_module(config, working_set)
     }
 
     fn call(
         &mut self,
         msg: Self::CallMessage,
         context: &Self::Context,
-        working_set: &mut WorkingSet<C>,
+        working_set: &mut WorkingSet<<Self::Context as Spec>::Storage>,
```
Comment (yaziciahmet) [inline crates/evm/src/lib.rs]: C as Spec

File: crates/citrea-stf/src/hooks_impl.rs [hunk 6]
```diff
@@ -61,7 +61,7 @@ impl<C: Context, Da: DaSpec> ApplySoftConfirmationHooks<Da> for Runtime<C, Da> {
     fn begin_soft_confirmation_hook(
         &mut self,
         soft_confirmation_info: &HookSoftConfirmationInfo,
-        working_set: &mut WorkingSet<Self::Context>,
+        working_set: &mut WorkingSet<<Self::Context as Spec>::Storage>,
```
Comment (yaziciahmet) [inline crates/citrea-stf/src/hooks_impl.rs]: C as Spec

File: bin/citrea/src/main.rs [hunk 7]
```diff
@@ -241,13 +249,16 @@ where
...
-        let rollup = CitreaRollupBlueprint::create_new_rollup(
+        let (mut rollup, rpc_methods) = CitreaRollupBlueprint::create_new_rollup(
             &rollup_blueprint,
             rt_genesis_paths,
             rollup_config,
         )
         .await
         .expect("Could not start full-node");
+
+        rollup.start_rpc_server(rpc_methods, None).await;
```
Comment (yaziciahmet) [inline bin/citrea/src/main.rs:260]: Is it on purpose that we don't return errors and panic from FullNode::start_rpc_server?

**Review by yaziciahmet — CHANGES_REQUESTED** [review]
Have only nits. Great cleanup PR.

File: crates/evm/src/query.rs (hunk 3 above)
Comment (eyusufatik) [reply crates/evm/src/query.rs:1760]: yes that's practically true for everywhere. BUT that's so much change everywhere, I don't know if I should continue :D 

basically %90 places where we pass Context, should be Storage actually

File: bin/citrea/src/main.rs (hunk 7 above)
Comment (eyusufatik) [reply bin/citrea/src/main.rs:260]: tbh, I don't even know

**Review by yaziciahmet — APPROVED** [review]

**Pull request merged**